### ISSUE TITLE
InvalidPlugin Error

### ISSUE THREAD

**Issue by MarcVCE** [noDRM/DeDRM_tools#68]
### Question / bug report

**InvalidPlugin**: The plugin in 'mydirectory' is invalid. It does not contain a top-level _init_py file

Version: Calibre 5.41
OS: Windows 10 LTSC
Python:  Python 3.8.10

I tried everything,still couldn't fix the error

### Which version of Calibre are you running?

5.41

### Which version of the DeDRM plugin are you running?

10.0.2 

### If applicable, which version of the Kindle software are you running?

None, just using Calibre

### Log output

**InvalidPlugin**: The plugin in 'mydirectory' is invalid.It does not contain a top-level __init__.py file

**Comment by ElleKayEm** [comment]
Looks like you're trying to load the wrong file.  Unzip DeDRM_tools_10.0.2.zip.  And load DeDRM_plugin.zip.

**Comment by MarcVCE** [comment]
> Looks like you're trying to load the wrong file. Unzip DeDRM_tools_10.0.2.zip. And load DeDRM_plugin.zip.

II just downloaded the DeDRM_tools-master.zip, should I go to the latest release and download the DeDRM_10.0.2.zip instead?
Just saying because the version was released the Nov 29, 2021, and you modified the [DeDRM_plugin 6 days ago.
Do the changes made in the main page applied to the DeDRM_10.0.2 released the Nov 29, 2021?

**Comment by ElleKayEm** [comment]
If you want to have the latest changes, you'll need to run makerelease.py.

**Comment by MarcVCE** [comment]
> If you want to have the latest changes, you'll need to run makerelease.py.

I see, but DeDRM_10.0.2.zip doesn't have a makerelease.py, file , do I copy the makerelease.py file form the main page, and paste it inside the  DeDRM_10.0.2.zip , then run it?
If, so where do I paste it? The DeDRM_10.0.2.zip, the DeDRM_plugin.zip inside the DeDRM_10.0.2.zip? Or where exactly?

Btw, I also tried to unDRM a book, and this error appeared:

ERROR: Failed to load the book: Couldn't open the book in 'directory of the book' - unknown.epub. Press «Show details» for more information.

Traceback (most recent call last):
  File "site-packages\calibre\gui2\viewer\ui.py", line 468, in _load_ebook_worker
  File "site-packages\calibre\gui2\viewer\convert_book.py", line 234, in prepare_book
  File "site-packages\calibre\gui2\viewer\convert_book.py", line 191, in do_convert
ConversionFailure: Failed to convert book: 'directory of the book' - unknown.epub with error:
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 428, in get_file_type_metadata
  File "site-packages\calibre\customize\builtins.py", line 194, in get_metadata
  File "site-packages\calibre\ebooks\metadata\epub.py", line 238, in get_quick_metadata
  File "site-packages\calibre\ebooks\metadata\epub.py", line 217, in get_metadata
  File "site-packages\calibre\ebooks\metadata\epub.py", line 164, in get_zip_reader
  File "site-packages\calibre\utils\localunzip.py", line 264, in __init__
  File "site-packages\calibre\utils\localunzip.py", line 240, in _extractall
ValueError: Not a ZIP file
DeDRM v10.0.2: Trying to decrypt book - unknown.epub
DeDRM v10.0.2: Verifying zip archive integrity
DeDRM v10.0.2: Error 'File is not a zip file' when checking zip archive
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 172, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 899, in run
  File "calibre_plugins.dedrm.__init__", line 278, in ePubDecrypt
  File "calibre_plugins.dedrm.zipfix", line 53, in __init__
  File "calibre_plugins.dedrm.zipfilerugged", line 685, in __init__
  File "calibre_plugins.dedrm.zipfilerugged", line 711, in _GetContents
  File "calibre_plugins.dedrm.zipfilerugged", line 723, in _RealGetContents
BadZipfile: File is not a zip file
InputFormatPlugin: EPUB Input running
on 'directory of the book' - unknown.epub
EPUB appears to be invalid ZIP file, trying a more forgiving ZIP parser
Failed to run pipe worker with command: from calibre.srv.render_book import viewer_main; viewer_main()
Python function terminated unexpectedly
  Not a ZIP file (Error Code: 1)
 Traceback (most recent call last):
  File "site.py", line 114, in main
  File "site.py", line 88, in run_entry_point
  File "site-packages\calibre\utils\ipc\worker.py", line 189, in main
  File "<string>", line 1, in <module>
  File "site-packages\calibre\srv\render_book.py", line 862, in viewer_main
  File "site-packages\calibre\srv\render_book.py", line 855, in render_for_viewer
  File "site-packages\calibre\srv\render_book.py", line 830, in render
  File "site-packages\calibre\ebooks\oeb\iterator\book.py", line 64, in extract_book
  File "site-packages\calibre\customize\conversion.py", line 246, in __call__
  File "site-packages\calibre\ebooks\conversion\plugins\epub_input.py", line 269, in convert
  File "site-packages\calibre\utils\localunzip.py", line 253, in extractall
  File "site-packages\calibre\utils\localunzip.py", line 240, in _extractall
ValueError: Not a ZIP file

Could you help me find out what is wrong?
Thanks a lot

**Comment by ElleKayEm** [comment]
DeDRM 10.0.2 is ready to go -- no need to run makerelease.py.  That is only if you want to create a zip file including the latest code here to load into calibre.  But as you said, 10.0.2 is from last November.

As far as the book you're trying to decrypt, is it an Adobe epub?  It doesn't seem to be recognized as a valid epub.

**Comment by MarcVCE** [comment]
Yep, I think so, I'm quite sure it is, how do you know if an epub is an Adobe epub?

**Comment by ElleKayEm** [comment]
Yes, this tool is for Adobe epubs (among other types of DRM).  How did you download it?  Usually you would download it by opening an acsm file with Adobe Digital Editions.

**Comment by MarcVCE** [comment]
I bought the book, and the only way to download it was from their webpage, also, the only way to view that book is from there app, it wasn't nothing to special or complicated, just a normal download from their webpage once I bought the book.
The booked must have been stored somewhere in my hard disk, so it wasn't difficult to find it, real difficulty is how to read it without using their damn app.

**Comment by ElleKayEm** [comment]
Probably a proprietory DRM then.

**Comment by MarcVCE** [comment]
Any possibility of bypassing proprietary DRM?

**Comment by ElleKayEm** [comment]
Not with this tool currently.

**Comment by MarcVCE** [comment]
> Not with this tool currently.
Do you know any tools that could help? 
Thank you very much for the info, I hope this projects develops to the point of being able to bypass one day, proprietary DRM

**Comment by ElleKayEm** [comment]
Where is the book from?

**Comment by MarcVCE** [comment]
Ridibooks

**Comment by ElleKayEm** [comment]
I'm not aware of anything that could help.

**Comment by MarcVCE** [comment]
No problem, thanks

**Comment by bangdosa93** [comment]
https://forum.novelupdates.com/threads/sad-news-riberty-for-ridibooks-stopped-working-how-to-remove-drm.128502/page-4#post-6816914

**Comment by MarcVCE** [comment]
> https://forum.novelupdates.com/threads/sad-news-riberty-for-ridibooks-stopped-working-how-to-remove-drm.128502/page-4#post-6816914

Didn't know that site, will look into it, thanks

**Comment by bangdosa93** [comment]
> > https://forum.novelupdates.com/threads/sad-news-riberty-for-ridibooks-stopped-working-how-to-remove-drm.128502/page-4#post-6816914
> 
> Didn't know that site, will look into it, thanks

tldr
there is at least one drm removal tool for the current windows application of ridibooks. (developer for this gone silent since October, but Ridi hasn't bothered to update their app since July or August) 
possibly a few others through telegram or other locked korean forum sites (being sold prbly)

**Comment by MarcVCE** [comment]
> > > https://forum.novelupdates.com/threads/sad-news-riberty-for-ridibooks-stopped-working-how-to-remove-drm.128502/page-4#post-6816914
> > 
> > 
> > Didn't know that site, will look into it, thanks
> 
> tldr there is at least one drm removal tool for the current windows application of ridibooks. (developer for this gone silent since October, but Ridi hasn't bothered to update their app since July or August, probably will if it gets a lot of exposure) possibly a few others through telegram or other locked korean forum sites (being sold prbly)

Has anyone found a way for other OS, I'm quite curious as I'm also a Linux user

**Comment by bangdosa93** [comment]
> > > > https://forum.novelupdates.com/threads/sad-news-riberty-for-ridibooks-stopped-working-how-to-remove-drm.128502/page-4#post-6816914
> > > 
> > > 
> > > Didn't know that site, will look into it, thanks
> > 
> > 
> > tldr there is at least one drm removal tool for the current windows application of ridibooks. (developer for this gone silent since October, but Ridi hasn't bothered to update their app since July or August, probably will if it gets a lot of exposure) possibly a few others through telegram or other locked korean forum sites (being sold prbly)
> 
> Has anyone found a way for other OS? I'm quite curious as I'm also a Linux user

there was a drm removal tool for previous versions of macos apps,  and the developer updated it once ( blogs' gone now so here's some web archive links)
http://web.archive.org/web/20210413021552/https://www.bpak.org/blog/2018/04/%EB%A6%AC%EB%94%94%EB%B6%81%EC%8A%A4-%EC%9E%90%EC%8B%A0%EC%9D%B4-%EC%86%8C%EC%9C%A0%ED%95%9C-%EC%B1%85-drm-%ED%95%B4%EC%A0%9C%ED%95%98%EA%B8%B0-feat-%EC%9C%84%ED%97%98%ED%95%9C-%EB%B9%84%EB%B0%80/

http://web.archive.org/web/20210413003751/https://www.bpak.org/blog/2019/05/%EB%A6%AC%EB%94%94%EB%B6%81%EC%8A%A4-drm-%ED%95%B4%EC%A0%9C%ED%95%98%EA%B8%B0-%ED%95%9C%EB%B2%88-%EB%8D%94-ft-riberty/

as for linux, I doubt anyone specifically made one for it, since ridi only has mac, windows and mobile apps

**Comment by MarcVCE** [comment]
> > > > > https://forum.novelupdates.com/threads/sad-news-riberty-for-ridibooks-stopped-working-how-to-remove-drm.128502/page-4#post-6816914
> > > > 
> > > > 
> > > > Didn't know that site, will look into it, thanks
> > > 
> > > 
> > > tldr there is at least one drm removal tool for the current windows application of ridibooks. (developer for this gone silent since October, but Ridi hasn't bothered to update their app since July or August, probably will if it gets a lot of exposure) possibly a few others through telegram or other locked korean forum sites (being sold prbly)
> > 
> > 
> > Has anyone found a way for other OS? I'm quite curious as I'm also a Linux user
> 
> there was a drm removal tool for previous versions of macos apps, and the developer updated it once ( blogs' gone now so here's some web archive links) http://web.archive.org/web/20210413021552/https://www.bpak.org/blog/2018/04/%EB%A6%AC%EB%94%94%EB%B6%81%EC%8A%A4-%EC%9E%90%EC%8B%A0%EC%9D%B4-%EC%86%8C%EC%9C%A0%ED%95%9C-%EC%B1%85-drm-%ED%95%B4%EC%A0%9C%ED%95%98%EA%B8%B0-feat-%EC%9C%84%ED%97%98%ED%95%9C-%EB%B9%84%EB%B0%80/
> 
> http://web.archive.org/web/20210413003751/https://www.bpak.org/blog/2019/05/%EB%A6%AC%EB%94%94%EB%B6%81%EC%8A%A4-drm-%ED%95%B4%EC%A0%9C%ED%95%98%EA%B8%B0-%ED%95%9C%EB%B2%88-%EB%8D%94-ft-riberty/
> 
> as for linux, I doubt anyone specifically made one for it, since ridi only has mac, windows and mobile apps

This fuckers tied things up well, what a hellish protection.

**Comment by alessionossa** [comment]
I am getting the same error.
